src/Framework/Hooks/CommonConditions.cs

```diff
@@ -197,20 +289,28 @@ private static bool ParseBool(string str)
                 return false;
 
             throw new InvalidCastException($"Unable to convert `{str}` to boolean.");
-        }
-
-        private static string GetCurrentWeatherName()
-        {
-            if (Game1.isRaining)
-                return "rainy";
-            if (Game1.isSnowing)
-                return "snowy";
-            if (Game1.isLightning)
-                return "stormy";
-            if (Game1.isDebrisWeather)
-                return "cloudy";
-
-            return "sunny";
-        }
-    }
-}
+        }
+
+        private static string GetCurrentWeatherName()
+        {
+            if (Game1.isRaining)
+                return "rainy";
+            if (Game1.isSnowing)
+                return "snowy";
+            if (Game1.isLightning)
+                return "stormy";
+            if (Game1.isDebrisWeather)
+                return "cloudy";
+
+            return "sunny";
+        }
+
+        private static bool DeprecatedCondition(Func<bool> conditionCallback)
+        {
+            string oldConditionName = "Friendship";
+            string newConditionName = "FriendshipLevel";
+            Monitor.Log($"`{oldConditionName}` is deprecated, use `{newConditionName}` instead", LogLevel.Warn);
+            return conditionCallback();
+        }

```

Comment:
I thouht something like this:

```suggestion
        private static bool DeprecatedCondition(Func<bool> conditionCallback, string oldConditionName, string newConditionName)
        {
            Monitor.Log($"`{oldConditionName}` is deprecated, use `{newConditionName}` instead", LogLevel.Warn);
            return conditionCallback();
        }
```

... for allow use this method repeatedly in future for other conditions which may deprecate in future.

Reply:
Changed